lib/core/proxy.c

```diff
@@ -557,21 +575,36 @@ static int on_1xx(h2o_httpclient_t *client, int version, int status, h2o_iovec_t
     return 0;
 }
 
-static void proceed_request(h2o_httpclient_t *client, size_t written, int is_end_stream)
+static void proceed_request(h2o_httpclient_t *client, size_t written, h2o_send_state_t send_state)
 {
     struct rp_generator_t *self = client->data;
+    if (self == NULL) {
+        return;
+    }
+    if (!h2o_send_state_is_in_progress(send_state)) {
```

Comment:
was the intention to detach the client on error here?

Reply:
yes